### ISSUE TITLE
Brain dump

### ISSUE THREAD

**Issue by AKuederle** [learn-and-code/MoneyPy#16]
Just a issue thread to write down some ideas. Feel free to use it :) If it grows to big, we will separate it into topic specfific brainstorm threads.

**Comment by AKuederle** [comment]
We will need some system to check if a certain entry already exists in the DB.
So far my ideas are:
1. The save method used to write the entries in the db, could get some sort of unique together parameterm which checks if a entrie already exists.
2. We could add a unique together constrained on db level, but depending on the datasource this might change.

**Comment by tweakimp** [comment]
We could create an md5 hash for every entry.

**Comment by AKuederle** [comment]
The issue with that is, that we might not want to include all the fields into this hash. E.g. a potential label field might have been modified by the user in the stored data, but not in the new data. But still the entries might be the same. A hash would be possible implementation of version 1. So you tell the method which fields should be included in the hash and then the hashs are calculated and compared

**Comment by tweakimp** [comment]
We just dont create a new hash for user changes. The hash is still standing for the same transaction although it is modified by the user. Can this lead to a problem? I cant think of one.

**Comment by AKuederle** [comment]
Good idea! Can't thing of an issue of the top of my head. Will think about
it a little more, but sounds like a solid option.

On Jan 31, 2018 1:29 PM, "FW" <notifications@github.com> wrote:

We just dont create a new hash for user changes. The hash is still standing
for the same transaction although it is modified by the user. Can this lead
to a problem? I cant think of one.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/learn-and-code/MoneyPy/issues/16#issuecomment-361918119>,
or mute the thread
